src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -3810,4 +3810,179 @@ BOOST_AUTO_TEST_CASE(TransposedConvolutionalLayerOptionalParameterTest)
     delete decoder;
 }
 
+BOOST_AUTO_TEST_CASE(BatchNormWithMinBatchesTest)
+{
+  arma::mat input, output, result, runningMean, runningVar, delta;
+
+  // Input consists of 3 images / outputs with 2 feature maps of size
+  // 4 x 1.
+  input = arma::mat(8, 3);
+  input << 1 << 446 << 42 << arma::endr
+      << 2 << 16 << 63 << arma::endr
+      << 3 << 13 << 63 << arma::endr
+      << 4 << 21 << 21 << arma::endr
+      << 1 << 13 << 11 << arma::endr
+      << 32 << 45 << 42 << arma::endr
+      << 22 << 16 << 63 << arma::endr
+      << 32 << 13 << 42 << arma::endr;
+
+  // Output calculated using torch.nn.BatchNorm2d().
+  result = arma::mat(8, 3);
+  result << -0.478614 << 3.2634 << -0.133844 << arma::endr
+      << -0.470205 << -0.352479 << 0.0427459 << arma::endr
+      << -0.461796 << -0.377706 << 0.0427459 << arma::endr
+      << -0.453387 << -0.310433 << -0.310433 << arma::endr
+      << -1.54289 << -0.84859 << -0.964306 << arma::endr
+      << 0.25072 << 1.00288 << 0.829303 << arma::endr
+      << -0.327864 << -0.675014 << 2.04433 << arma::endr
+      << 0.25072 << -0.84859 << 0.829303 << arma::endr;
```

Comment:
These lines have some discrepancy.

Reply:
If you have spare time, fill in the exact same values from PyTorch and set error to 0.1 (same as BatchNormTest) and it will pass the test.

Reply:
Yep! I checked. That works perfectly, thanks.